sound/soc/sof/intel/Kconfig

```diff
@@ -111,17 +111,37 @@ config SND_SOC_SOF_HDA_COMMON
 	tristate
 	select SND_SOC_SOF_PCI
 	select SND_SOC_ACPI_INTEL_MATCH
+	select SND_SOC_SOF_HDA_LINK_BASELINE
 
-config SND_SOC_SOF_HDA
-	tristate "SOF support for HDA Links(HDA/HDMI)"
-	depends on SND_SOC_SOF_HDA_COMMON
-	select SND_HDA_EXT_CORE
-	select SND_SOC_HDAC_HDA
-	select SND_SOC_HDAC_HDMI
+if SND_SOC_SOF_HDA_COMMON
+
+config SND_SOC_SOF_HDA_LINK
+	bool "SOF support for HDA Links(HDA/HDMI)"
 	help
           This adds support for HDA links(HDA/HDMI) with Sound Open Firmware
 		  for Intel(R) platforms.
           Say Y if you want to enble HDA links with SOF.
           If unsure select "N".
 
+if SND_SOC_SOF_HDA_LINK
+config SND_SOC_SOF_HDA_AUDIO_CODEC
+	bool "SOF support for HDAudio codecs"
+	help
+          This adds support for HDAudio codecs with Sound Open Firmware
+		  for Intel(R) platforms.
+          Say Y if you want to enble HDAudio codecs with SOF.
+          If unsure select "N".
+endif ## SND_SOC_SOF_HDA_LINK
+
+endif ## SND_SOC_SOF_HDA_COMMON
+
+config SND_SOC_SOF_HDA_LINK_BASELINE
+	tristate
+	select SND_SOC_SOF_HDA if SND_SOC_SOF_HDA_LINK
+
+config SND_SOC_SOF_HDA
+	tristate
+	select SND_HDA_EXT_CORE if SND_SOC_SOF_HDA_LINK
+	select SND_SOC_HDAC_HDA if SND_SOC_SOF_HDA_AUDIO_CODEC
+
```

Comment:
@keyonjie  @plbossart  sorry, one  I find SND_SOC_HDAC_HDMI and SND_SOC_HDAC_HDA is selected in machine driver's kconfig.  And SND_HDA_EXT_CORE is selected by SND_SOC_HDAC_HDMI. Why we need to select them at here?

Reply:
There is a difference between HDAC_HDA and HDAC_HDMI.
In the first case, the SST/SOF drivers explicitly reference the io_ops, so the hdac_hda code needs to be compiled in the same way as the SOF/SST drivers. If they latter are built-in, hdac_hda needs to be built-in as well. That's why there is a select from the platform driver side, it models the code dependency.
There is no such dependency for HDAC_HDMI, so the parameter is not selected directly. It will be selected by a machine driver if needed, and it can be either built-in or module.
Does this help understand the difference?